src/main/java/ch/njol/skript/lang/Variable.java

```diff
@@ -578,14 +572,15 @@ public void change(Event event, @Nullable Object[] delta, ChangeMode mode) throw
 					if (mode == ChangeMode.REMOVE) {
 						if (map == null)
 							return;
-						ArrayList<String> toRemove = new ArrayList<>(); // prevents CMEs
+						Set<String> toRemove = new HashSet<>(); // prevents CMEs
 						for (Object value : delta) {
 							for (Entry<String, Object> entry : map.entrySet()) {
+								String key = entry.getKey();
+								if (key == null)
+									continue; // This is NOT a part of list variable
+								if (toRemove.contains(key))
+									continue; // Skip if we've already marked this key to be removed
 								if (Relation.EQUAL.isImpliedBy(Comparators.compare(entry.getValue(), value))) {
```

Comment:
wouldn't it be much more efficient to reverse the loops and have the delta loop be the inner loop?

Reply:
No because it'd act like `REMOVE_ALL`, checking every delta value for each key. Unless we convert delta into a list and modify it if a key is removed, I don't know if that's really worth it though. what do you think?

Reply:
current:
```
for each value to remove:
    check every value in the list:
        if list value is null, or already removed, skip to next in list
        if remove and list values equal, remove from list
```  
proposed:
```
for each value in the list:
    if list value is null, or already removed, skip to next in list
    check value to remove:
        if remove and list values equal, remove from list
```  
Wouldn't this be the same behavior?

Reply:
oh i forgot the break

Reply:
yes i suppose it's fine then
you could pre-emptively remove nulls before the loops if you wanted, but not sure if worthwhile.